### PR TITLE
OADP-1057: AWS STS Creds support for internal image backup

### PR DESCRIPTION
**Pull request by kaovilai**
- add support for aws sts creds for registry
- remove image-spec replace breaking build, previously there for CVE.. now the dep is much newer..
- [OADP-1057](https://issues.redhat.com//browse/OADP-1057): STS creds support for internal image backup
- [remove kopia dependency by removing podvolume package](https://github.com/openshift/openshift-velero-plugin/pull/200) also available as a separate PR. #200  but it is required here to build.. so can rebase after that merged or vice versa.

#### ⏯️ [Testing instructions](https://github.com/openshift/oadp-operator/pull/1126#:~:text=Testing%20instructions) 🏈 

TODO:
- [x] remove `replace github.com/vmware-tanzu/velero v1.11.0 => github.com/kaovilai/velero v0000- // TODO: remove when https://github.com/openshift/velero/pull/279 is merged` https://github.com/openshift/velero/pull/279
- [x] Update secret handling to support new field

test image
```
ghcr.io/kaovilai/openshift-velero-plugin:sts
```

dpa
```
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
  name: dpa-sample
spec:
  unsupportedOverrides:
    openshiftPluginImageFqin: ghcr.io/kaovilai/openshift-velero-plugin:sts
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -107,109 +89,121 @@ require (
...
+)
 
 // CVE-2021-3121
 replace github.com/gogo/protobuf => github.com/gogo/protobuf v1.3.2
 
 replace bitbucket.org/ww/goautoneg v0.0.0-20120707110453-75cd24fc2f2c => github.com/markusthoemmes/goautoneg v0.0.0-20190713162725-c6008fefa5b1
 
-replace github.com/vmware-tanzu/velero => github.com/openshift/velero v0.10.2-0.20230320194820-cd848ef1a58f
+// replace github.com/vmware-tanzu/velero v1.11.0 => github.com/openshift/velero v0.10.2-0.20230320194820-cd848ef1a58f
+replace github.com/vmware-tanzu/velero v1.11.0 => github.com/kaovilai/velero v1.6.1-0.20230711234837-eef3a4de9b80 // TODO: remove when https://github.com/openshift/velero/pull/277 is merged
```
Comment (kaovilai) [inline go.mod]: TODO: Correct this when https://github.com/openshift/velero/pull/277 is merged.

**Comment by kaovilai** [comment]
Did some sanity testing. The changes work as intended. Marking ready for review.

**Comment by weshayutin** [comment]
w/ backupImages: true
w/ patch:

```
https://termbin.com/0h91

[whayutin@thinkdoe ROSA]$ velero backup get hello-world-patch2 
NAME                 STATUS      ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION     SELECTOR
hello-world-patch2   Completed   0        0          2023-07-24 18:38:14 +0000 UTC   29d       wdh07052023d-dpa-1   <none>
[whayutin@thinkdoe ROSA]$ oc get dpa -o yaml | grep -i image
    backupImages: true
      openshiftPluginImageFqin: ghcr.io/kaovilai/openshift-velero-plugin:sts
```

**Review by sseago — APPROVED** [review]

**Pull request merged**